mitiq/calibration/calibrator.py

```diff
@@ -38,7 +26,52 @@
     Observable,
     QuantumResult,
 )
-from mitiq.calibration.settings import Settings, Strategy, DefaultStrategy
+from mitiq.calibration.settings import (
+    Settings,
+    Strategy,
+    BenchmarkProblem,
+)
+
+
+class ExperimentResults:
+    """Class to store calibration experiment data, and provide helper methods
+    for computing results based on it."""
+
+    def __init__(self, num_strategies: int, num_problems: int) -> None:
+        self.num_strategies = num_strategies
+        self.num_problems = num_problems
+        self.mitigated = np.full((num_strategies, num_problems), np.nan)
+        self.noisy = np.full((num_strategies, num_problems), np.nan)
+        self.ideal = np.full((num_strategies, num_problems), np.nan)
+
+    def add_result(
+        self,
+        strategy: Strategy,
+        problem: BenchmarkProblem,
+        *,
+        ideal_val: float,
+        noisy_val: float,
+        mitigated_val: float,
+    ) -> None:
+        """Add a single result from a (Strategy, BenchmarkProblem) pair and
+        store the results."""
+        self.mitigated[strategy.id, problem.id] = mitigated_val
+        self.noisy[strategy.id, problem.id] = noisy_val
+        self.ideal[strategy.id, problem.id] = ideal_val
+
+    def squared_errors(self) -> npt.NDArray[np.float32]:
+        """Returns an array of squared errors, one for each (strategy, problem)
+        pair."""
+        return (self.ideal - self.mitigated) ** 2
+
```

Comment:
Here we can add a new method `check_if_full()` with something like:

```
 if np.isnan(self.results.mitigated).any():
          raise ValueError( "One or more results from calibration experiments are missing. ")
```
such that we can call it in multiple places:

- Inside `best_strategy_id()`. Which I think is the most critical place to avoid the risk of false best strategies. 

- And, if you think it's useful, somewhere in the `Calibrator` . E.g., where you have currently placed this check.

Moreover, it's probably good to have a unit **test** for triggering this error.